keepercommander/commands/folder.py

```diff
@@ -253,47 +251,65 @@ def execute(self, params, **kwargs):
                 raise CommandError('ls', '{0}: No such folder or record'.format(pattern))
         else:
             if show_detail:
-                if len(folders) > 0:
-                    table = []
-                    headers = ['folder_uid', 'name', 'flags', 'parent_uid']
+                if fmt in ('json', 'csv'):
+                    combined_table = []
+                    combined_headers = ['type', 'uid', 'name', 'details']
+                    
+                    if len(folders) > 0:
+                        def folder_flags(f):
+                            if f.type == 'shared_folder':
+                                flags = 'S'
+                            else:
+                                flags = ''
+                            return flags
+                        
+                        for f in folders:
+                            row = ['folder', f.uid, f.name, f'Flags: {folder_flags(f)}, Parent: {f.parent_uid or "/"}']
+                            combined_table.append(row)
+                    
+                    if len(records) > 0:
+                        for record in records:
+                            row = ['record', record.record_uid, record.title, 
+                                   f'Type: {record.record_type}, Description: {vault_extensions.get_record_description(record)}']
+                            combined_table.append(row)
+                    
+                    combined_table.sort(key=lambda x: (x[2] or '').lower())
```

Comment:
This sort can be accompanied by group by
```suggestion
                    combined_table.sort(key=lambda x: (x[0], (x[2] or '').lower()))
```

Reply:
Resolved